testing-proto/src/test/java/io/grpc/testing/protobuf/SimpleServiceTest.java

```diff
@@ -57,4 +72,90 @@ public void serviceMethodDescriotrs() {
   }
+
+  public static class AnnotationProcessor extends AbstractProcessor {
+
+    private boolean processedClass = false;
+
+    @Override
+    public Set<String> getSupportedAnnotationTypes() {
+      return Collections.singleton(RpcMethod.class.getCanonicalName());
+    }
+
+    private void verifyRpcMethodAnnotation(
+        MethodDescriptor<SimpleRequest, SimpleResponse> descriptor, RpcMethod annotation) {
+      assertEquals(descriptor.getFullMethodName(), annotation.fullMethodName());
+      assertEquals(descriptor.getType(), annotation.methodType());
+
+      // Class objects may not be available at runtime, handle MirroredTypeException if it occurs
+      try {
+        assertEquals(SimpleRequest.class, annotation.requestType());
+      } catch (MirroredTypeException e) {
+        assertEquals(SimpleRequest.class.getCanonicalName(), e.getTypeMirror().toString());
+      }
+
+      try {
+        assertEquals(SimpleResponse.class, annotation.responseType());
+      } catch (MirroredTypeException e) {
+        assertEquals(SimpleResponse.class.getCanonicalName(), e.getTypeMirror().toString());
+      }
+    }
+
+    @Override
+    public synchronized boolean process(
+        Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+      for (Element rootElement : roundEnv.getRootElements()) {
+        if (!rootElement.asType().toString().equals(SimpleServiceGrpc.class.getCanonicalName())) {
+          continue;
+        }
+
+        Map<String, RpcMethod> methodToAnnotation = new HashMap<String, RpcMethod>();
+        for (Element enclosedElement : rootElement.getEnclosedElements()) {
+          RpcMethod annotation = enclosedElement.getAnnotation(RpcMethod.class);
+          if (annotation != null) {
+            methodToAnnotation.put(enclosedElement.getSimpleName().toString(), annotation);
+          }
+        }
+
+        verifyRpcMethodAnnotation(
+            SimpleServiceGrpc.getUnaryRpcMethod(), methodToAnnotation.get("getUnaryRpcMethod"));
+        verifyRpcMethodAnnotation(
+            SimpleServiceGrpc.getServerStreamingRpcMethod(),
+            methodToAnnotation.get("getServerStreamingRpcMethod"));
+        verifyRpcMethodAnnotation(
+            SimpleServiceGrpc.getClientStreamingRpcMethod(),
+            methodToAnnotation.get("getClientStreamingRpcMethod"));
+        verifyRpcMethodAnnotation(
+            SimpleServiceGrpc.getBidiStreamingRpcMethod(),
+            methodToAnnotation.get("getBidiStreamingRpcMethod"));
+
+        processedClass = true;
+      }
+      return false;
+    }
+  }
+
+  @Test
+  public void testRpcMethodAnnotations() throws Exception {
+    JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+    StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
+    AnnotationProcessor processor = new AnnotationProcessor();
+
+    Iterable<? extends JavaFileObject> obs =
```

Comment:
ideally this would be stored as a resource for the test.  I think our windows build may not like this test.

As a work around, maybe you can add an `Assume` call at the beginning to assert the file exists?

Reply:
Added an Assume.